WalletWasabi.Fluent/ViewModels/Wallets/Labels/SuggestionLabelsViewModel.cs

```diff
@@ -13,143 +13,68 @@ namespace WalletWasabi.Fluent.ViewModels.Wallets.Labels;
 
-	public KeyManager KeyManager { get; }
-	public Intent Intent { get; }
-
-	public void UpdateLabels()
-	{
-		var labelPool = new Dictionary<string, int>(); // int: score.
-
-		// Make recent and receive labels count more for the current wallet
-		var multiplier = 100;
-		foreach (var label in KeyManager.GetReceiveLabels().Reverse().SelectMany(x => x.Labels))
-		{
-			var score = (Intent == Intent.Receive ? 100 : 1) * multiplier;
-			if (!labelPool.TryAdd(label, score))
-			{
-				labelPool[label] += score;
-			}
-
-			if (multiplier > 1)
-			{
-				multiplier--;
-			}
-		}
-
-		// Receive addresses should be more dominant.
-		foreach (var label in WalletHelpers.GetReceiveAddressLabels().SelectMany(x => x.Labels))
-		{
-			var score = Intent == Intent.Receive ? 100 : 1;
-			if (!labelPool.TryAdd(label, score))
-			{
-				labelPool[label] += score;
-			}
-		}
-
-		// Change addresses shouldn't be much dominant, but should be present.
-		foreach (var label in WalletHelpers.GetChangeAddressLabels().SelectMany(x => x.Labels))
-		{
-			var score = 1;
-			if (!labelPool.TryAdd(label, score))
-			{
-				labelPool[label] += score;
-			}
-		}
-
-		multiplier = 100; // Make recent labels count more.
-		foreach (var label in WalletHelpers.GetTransactionLabels().SelectMany(x => x.Labels).Reverse())
-		{
-			var score = (Intent == Intent.Send ? 100 : 1) * multiplier;
-			if (!labelPool.TryAdd(label, score))
-			{
-				labelPool[label] += score;
-			}
-
-			if (multiplier > 1)
-			{
-				multiplier--;
-			}
-		}
-
-		var unwantedLabelSuggestions = new[]
-		{
-			"test", // Often people use the string "test" as a label. It obviously cannot be a real label, just a test label.
-			"zerolink mixed coin", // Obsolated autogenerated label from old WW1 versions.
-			"zerolink change", // Obsolated autogenerated label from old WW1 versions.
-			"zerolink dequeued change" // Obsolated autogenerated label from old WW1 versions.
-		};
-
-		var labels = labelPool
-			.Where(x =>
-				!unwantedLabelSuggestions.Any(y => y.Equals(x.Key, StringComparison.OrdinalIgnoreCase))
-				&& !x.Key.StartsWith("change of (", StringComparison.OrdinalIgnoreCase)); // An obsolated autogenerated label pattern was from old WW1 versions starting with "change of (".
-
-		var mostUsedLabels = labels
-			.GroupBy(x => x.Key)
-			.Select(x => new
-			{
-				Label = x.Key,
-				Score = x.Sum(y => y.Value)
-			})
-			.OrderByDescending(x => x.Score)
-			.ToList();
-
-		_sourceLabels.Clear();
-		_sourceLabels.AddRange(
-			mostUsedLabels.Select(x => new SuggestionLabelViewModel(x.Label, x.Score)));
-	}
+	private KeyManager KeyManager { get; }
+	private Intent Intent { get; }
 
 	private void CreateSuggestions(int topSuggestionsCount)
 	{
+		var ranking = LabelRanking.Rank(new RankInput(KeyManager.GetReceiveLabels(), WalletHelpers.GetReceiveAddressLabels(), WalletHelpers.GetChangeAddressLabels(), WalletHelpers.GetTransactionLabels()), Intent);
```

Comment:
`RankInput` class feels unnecessary... as either way you have to pass 4 parameters here. So you could pass them straightaway to the method.